ls1intum/Artemis
# Feature/Programming Exercise/Trigger buttons should detect manual results and passed due dates

### Checklist

### Motivation and Context



Fixes #1014 

### Description


- Single build trigger button: Check if the deadline has passed and if the participation's latest result is manual -> show warning modal
- Trigger all builds button: Check if the deadline has passed -> show warning modal

### Steps for Testing


Given a programming exercise that has a dueDate / buildAndTestAfterDueDate that has passed:
1. Log in to Artemis
2. Navigate to Course Administration > Programming Exercise > Participations
3. Click the trigger button for a single participation that has a manual result as its latest result => a modal should warn the instructor 

1. Log in to Artemis
2. Navigate to Course Administration > Programming Exercise > Participations
3. Click the trigger all button => a modal should warn the instructor 

### Screenshots



![trigger-warning](https://user-images.githubusercontent.com/28230611/68591529-3ab60d80-0491-11ea-994a-059e479a5045.gif)
